tests/test_repository_pypi.py

```diff
@@ -176,3 +179,18 @@ def test_wheel_cache_cleanup_called(
     ireq = from_line("six==1.10.0")
     pypi_repository.get_dependencies(ireq)
     WheelCache.return_value.cleanup.assert_called_once_with()
+
+
+@mock.patch("piptools.repositories.pypi.PyPIRepository.resolve_reqs")  # to run offline
+@mock.patch("piptools.repositories.pypi.WheelCache")
+def test_relative_path_cache_dir(WheelCache, resolve_reqs, from_line):
```

Comment:
Consider my suggestions to simplify the test. Also, I've added a test for pip's cache dir.
See https://github.com/kammala/pip-tools/pull/1

Reply:
Thanks for suggestion, merged your changes!

I was not sure about using protected attributes in tests(seems to be a little bit fragile for me)

Reply:
I think it's fine, nothing wrong with that in unit tests. Thanks for merging my suggestions!